lib/Promise/ES6.pm

```diff
@@ -255,18 +255,29 @@ sub new {
     my $resolver = sub {
         $$value_sr = $_[0];
         bless $value_sr, _RESOLUTION_CLASS();
-        _propagate_if_needed( $value_sr, \@children );
+
+        if ( ref $$value_sr eq __PACKAGE__ ) {
```

Comment:
Ohh ok.

This is still wrong because it needs to be isa(), in case we have a subclass.